research/src/fetch_campaign_donations.py

```diff
@@ -0,0 +1,289 @@
+                    'Número Recibo Eleitoral': 'electoral_receipt_number',
+                    'Número UE': 'electoral_unit_number',
+                    'Sequencial Candidato': 'candidate_sequence',
+                    'Sequencial prestador conta': 'accountable_sequence',
+                    'Sequencial comite': 'committee_sequence',
+                    'Sequencial Diretorio': 'party_board_sequence',
+                    'Setor econômico do doador': 'donor_economic_sector',
+                    'Setor econômico do doador originário':
+                    'original_donor_economic_sector',
+                    'Sigla da UE': 'electoral_unit_abbreviation',
+                    'Sigla Partido': 'party_acronym',
+                    'Sigla UE doador': 'donor_electoral_unit_abbreviation',
+                    'Tipo de documento': 'document_type',
+                    'Tipo diretorio': 'party_board_type',
+                    'Tipo doador originário': 'original_donor_type',
+                    'Tipo partido': 'party_type',
+                    'Tipo receita': 'revenue_type',
+                    'Tipo comite': 'committee_type',
+                    'UF': 'state',
+                    'Valor receita': 'revenue_value'}
+    if 'candidates' in donations_data.keys():
+        donations_data['candidates'].rename(columns=translations,
+                                            inplace=True)
+    if 'parties' in donations_data.keys():
+        donations_data['parties'].rename(columns=translations,
+                                         inplace=True)
+    if 'committees' in donations_data.keys():
+        donations_data['committees'].rename(columns=translations,
+                                            inplace=True)
+
+
+def strip_columns_names(donations_data, key):
+    if key in donations_data.keys():
+        columns_candidates = donations_data[key].columns.values
+        columns_mod_candidates = {}
+        for name in columns_candidates:
+            columns_mod_candidates[name] = name.strip()
+        donations_data[key].rename(columns=columns_mod_candidates,
+                                   inplace=True)
+
+
+def download_base(url, year):
+    file_name = url.split('/')[-1]
+    print("Downloading " + file_name)
+    urllib.request.urlretrieve(url, file_name, reporthook)
+    print("Uncompressing downloaded data.")
+    with zipfile.ZipFile(file_name, "r") as zip_ref:
+        zip_ref.extractall(file_name.split('.')[0])
+    os.remove(file_name)
+
+    print("Reading all the data and creating the dataframes...")
+
+    donations_data = folder_walk(year)
+
+    strip_columns_names(donations_data, 'candidates')
+    strip_columns_names(donations_data, 'parties')
+    strip_columns_names(donations_data, 'committees')
+
+    return donations_data
+
+
+if __name__ == "__main__":
+    base_url = ('http://agencia.tse.jus.br/estatistica/sead/odsele/'
+                'prestacao_contas/')
+    url = base_url + 'prestacao_contas_2010.zip'
```

Comment:
Can we do something like I said before like use the same structure but `format` to the years, using an apply for a dictionary with years?

@cuducos can explain better here